Neos.Flow/Classes/Cli/RequestBuilder.php

```diff
@@ -214,10 +220,44 @@ protected function parseRawCommandLineArguments(array $rawCommandLineArguments,
                 }
                 $argumentName = $this->extractArgumentNameFromCommandLinePart($rawArgument);
 
+                $parameterNameForArrayType = $this->findParameterNameForArrayArgument(
+                    $argumentName,
+                    $optionalArguments,
+                    $requiredArguments
+                );
+                $pathAsArray = [];
+                $matches = [];
+                // if a command controller parameter name with type array is found, parse the dot syntax
+                if (!empty($parameterNameForArrayType) &&
+                    0 < \preg_match_all('/([^\."\']+)|"([^"]*)"|\'([^\']*)\'/', $argumentName, $matches)
```

Comment:
```suggestion
                    0 < \preg_match_all('/([^\.\[]+)|\[([^\]]*)\]/', $argumentName, $matches)
```

... to parse array keys with `--myarray[...]`-syntax